### PR TITLE
TMEDIA 280 replace corecomponents package, error handling story feed tag

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Copy-paste corecomponents package
Add error handling for falsy tag slug

## Jira Ticket
- [TMEDIA-280](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-280)

## Acceptance Criteria
The Story Feed Tags Content Source block no longer has arc-core-components as a dependency

NOTE: Make sure it’s also removed from the block’s package.json!

If no tag is provided, the content source does not make a request to the content api.
## Test Steps

1. Checkout this branch `git checkout TMEDIA-280-story-feed-tag-perf`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/story-feed-tag-content-source-block `
3. Use content debugger and see preexisting request still working like "sports" tag
Screen Shot 2021-07-26 at 09 46 41
4. Throw error if falsy param for tag slug 

## Dependencies or Side Effects
- removing deps 

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+*/
+const pattern = (key = {}) => {
+  const website = key['arc-site'] || 'Arc Site is not defined';
+  const { tagSlug, feedOffset, feedSize } = key;
+
+  if (!tagSlug) {
+    throw new Error('tagSlug parameter is required');
+  }
+
+  return `/content/v4/search/published?q=taxonomy.tags.slug:${tagSlug}&size=${feedSize}&from=${feedOffset}&sort=display_date:desc&website=${website}`;
```
Comment (matthewroach) [inline blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js]: `feedSize` and `feedOffset` do not have a default, which causes an error. In my PR I added defaults here to account for this

**Review by matthewroach — CHANGES_REQUESTED** [review]
Would be nice to add defaults for `feedSize` and `feedOffset`

File: blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js (hunk 1 above)
Comment (JackHowa) [reply blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js]: will do thanks

Comment (JackHowa) [reply blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js]: updated @matthewroach

**Review by matthewroach — APPROVED** [review]

File: blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js (hunk 1 above)
Comment (JackHowa) [reply blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js]: before 

throws error on tag slug but no feedSize or feedoffset

after 

Screen Shot 2021-07-26 at 10 51 25

**Pull request merged**